packages/@webex/internal-plugin-metrics/src/metrics.types.ts

```diff
@@ -17,7 +17,7 @@ export interface ClientEvent {
 
 export interface BehavioralEvent {
   // TODO: not implemented
-  name: never;
+  name: 'host.meeting.participant.admitted' | 'client.pageJMT.received' | 'sdk.media-flow.started';
   payload?: never;
   options?: never;
```

Comment:
if you do not have a common interface for the events then you do not need to add `payload` and `options`, same for the `InternalEvent`

Reply:
also we should have a type defined as a union of all these interfaces, something like this:
`export type Event = ClientEvent | BehavioralEvent | OperationalEvent | FeatureEvent | MediaQualityEvent;`
and use that type in functions like submitClientEvent()

Reply:
I would like to keep the API consistent for all of them, and I think they will do require name, payload and options in future iterations.

Also, I have not created a union type with all the types as I don't need it. Each behaviour type will have different types for payload, and options, and I want to keep that separate. I only want the structure of the functions to be the same.

Reply:
ok, so if we always have separate functions for sending each kind of event (like client event, behavioural, feature etc) then OK we don't need the union